### PR TITLE
[hlm-1373]: household

### PR DESCRIPTION
**Pull request by shivab-egov**

### CODE REVIEW

File: health-services/household/pom.xml [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-jdbc</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.egov.common</groupId>
+            <artifactId>health-services-common</artifactId>
+            <version>1.0.7-SNAPSHOT</version>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
```
Comment (jayantp-egov) [inline health-services/household/pom.xml:50]: Remove redis dependency! Already there in common now.

File: health-services/household/src/main/java/org/egov/HouseholdApplication.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package org.egov;
+
+import org.egov.tracer.config.TracerConfiguration;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.cache.annotation.EnableCaching;
+import org.springframework.context.annotation.Import;
+@Import({ TracerConfiguration.class })
```
Comment (jayantp-egov) [inline health-services/household/src/main/java/org/egov/HouseholdApplication.java:8]: Put a new line here

File: health-services/household/src/main/java/org/egov/service/HouseholdService.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    @Autowired
+    public HouseholdService(HouseholdRepository householdRepository, IdGenService idGenService) {
+        this.householdRepository = householdRepository;
+        this.idGenService = idGenService;
+    }
+
+    public List<Household> create(HouseholdRequest householdRequest) throws Exception {
+        List<String> ids = householdRequest.getHousehold().stream().map(Household::getClientReferenceId)
+                .filter(Objects::nonNull).collect(Collectors.toList());
+        if (!ids.isEmpty()) {
```
Comment (jayantp-egov) [inline health-services/household/src/main/java/org/egov/service/HouseholdService.java:47]: No need to check existing for create

File: health-services/household/src/main/java/org/egov/service/HouseholdService.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+        householdRepository.save(householdRequest.getHousehold(), "save-household-topic");
+        return householdRequest.getHousehold();
+    }
+
+    public List<Household> search(HouseholdSearchRequest request, Integer limit, Integer offset, String tenantId,
+                                  Long lastChangedSince, Boolean includeDeleted) throws QueryBuilderException {
+
+        if (isSearchByIdOnly(request.getHousehold())) {
+            List<String> ids = new ArrayList<>();
```
Comment (jayantp-egov) [inline health-services/household/src/main/java/org/egov/service/HouseholdService.java:77]: Can we make this common?

File: health-services/household/src/main/java/org/egov/web/models/Household.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+@javax.annotation.Generated(value = "org.egov.codegen.SpringBootCodegen", date = "2022-12-21T13:41:16.379+05:30")
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+public class Household {
+
+    @JsonProperty("id")
+    private String id = null;
```
Comment (jayantp-egov) [inline health-services/household/src/main/java/org/egov/web/models/Household.java:29]: Put min max here

**Review by jayantp-egov — CHANGES_REQUESTED** [review]

File: health-services/household/pom.xml (hunk 1 above)
Comment (shivab-egov) [reply health-services/household/pom.xml]: Updated.

File: health-services/household/src/main/java/org/egov/HouseholdApplication.java (hunk 2 above)
Comment (shivab-egov) [reply health-services/household/src/main/java/org/egov/HouseholdApplication.java:9]: Done.

File: health-services/household/src/main/java/org/egov/service/HouseholdService.java (hunk 3 above)
Comment (shivab-egov) [reply health-services/household/src/main/java/org/egov/service/HouseholdService.java]: Done.

File: health-services/household/src/main/java/org/egov/web/models/Household.java (hunk 5 above)
Comment (shivab-egov) [reply health-services/household/src/main/java/org/egov/web/models/Household.java:31]: Done.

File: health-services/household/src/main/java/org/egov/service/HouseholdService.java (hunk 4 above)
Comment (shivab-egov) [reply health-services/household/src/main/java/org/egov/service/HouseholdService.java:70]: Done.

**Pull request merged**